include/seqan3/argument_parser/validators.hpp

```diff
@@ -297,68 +290,347 @@ class file_ext_validator
+    using file_validator_base::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
      */
-    template <std::ranges::ForwardRange range_type>
-    //!\cond
-        requires std::ConvertibleTo<value_type_t<range_type>, std::filesystem::path const &>
-    //!\endcond
-    void operator()(range_type const & v) const
+    output_directory_validator() = default;                                               //!< Defaulted.
+    output_directory_validator(output_directory_validator const &) = default;             //!< Defaulted.
+    output_directory_validator(output_directory_validator &&) = default;                  //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator const &) = default; //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator &&) = default;      //!< Defaulted.
+    ~output_directory_validator() = default;                                              //!< Defaulted.
+
+    //!\brief Inherit base constructor.
+    using file_validator_base::file_validator_base;
+    //!\}
+
+    // Import the base::operator().
+    using file_validator_base::operator();
+
+    /*!\brief Tests whether path is writable.
+     * \param dir The input value to check.
+     * \throws parser_invalid_argument if the validation process failed or std::filesystem::filesystem_error if some
+     *         unexpected errors during the filesystem operations occur.
+     */
+    void operator()(std::filesystem::path const & dir) const override
     {
-         std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+        std::function<void(std::filesystem::path const *)> deleter;
+        if (std::filesystem::exists(dir))
+            deleter = [] (std::filesystem::path const *) { /* no-op */ };
+        else
+            deleter = [] (std::filesystem::path const * p)
+            {
+                [[maybe_unused]] std::error_code ec;
+                std::filesystem::remove(*p, ec);
+            };
+
+        // Make sure the created dir is deleted after we are done.
+        std::unique_ptr<std::filesystem::path const, decltype(deleter)> raii{&dir, deleter};
+
+        std::error_code ec;
+        std::filesystem::create_directory(dir, ec); // does nothing and is not treated as error if path already exists.
+        // if error code was set or if dummy.txt could not be created within the output dir, throw an error.
+        if (static_cast<bool>(ec) || !path_is_writable(dir / "dummy.txt"))
+        {
+            throw parser_invalid_argument(detail::to_string("Cannot create directory: ", dir, "!"));
+        }
     }
 
     //!\brief Returns a message that can be appended to the (positional) options help page info.
     std::string get_help_page_message() const
     {
-        return detail::to_string("The file or directory is checked for existence.");
+        return detail::to_string("An output directory.");
```

Comment:
```suggestion
        return detail::to_string("The output directory must be writable.");
```

Reply:
What about: `A valid path for the output directory.` 
I feel this fits better to describe the limit of this option. I mean output directory implies that it is writable.